resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/kubernetes/integrationtest/KubernetesSuite.scala

```diff
@@ -282,7 +273,42 @@ private[spark] class KubernetesSuite extends SparkFunSuite with BeforeAndAfter {
       "Unexpected value for annotation2")
   }
 
-  test("Enable SSL on the driver submit server") {
+  test("Enable SSL on the driver submit server using stores") {
+    val (keyStore, trustStore) = SSLUtils.generateKeyStoreTrustStorePair(
+      Minikube.getMinikubeIp,
+      "changeit",
+      "changeit",
+      "changeit")
+    val args = Array(
+      "--master", s"k8s://https://${Minikube.getMinikubeIp}:8443",
+      "--deploy-mode", "cluster",
+      "--kubernetes-namespace", NAMESPACE,
+      "--name", "spark-pi",
+      "--executor-memory", "512m",
+      "--executor-cores", "1",
+      "--num-executors", "1",
+      "--jars", HELPER_JAR_FILE.getAbsolutePath,
+      "--class", SPARK_PI_MAIN_CLASS,
+      "--conf", s"spark.kubernetes.apiserver.caCertFile=${clientConfig.getCaCertFile}",
+      "--conf", s"spark.kubernetes.apiserver.clientKeyFile=${clientConfig.getClientKeyFile}",
+      "--conf", s"spark.kubernetes.apiserver.clientCertFile=${clientConfig.getClientCertFile}",
+      "--conf", "spark.kubernetes.executor.docker.image=spark-executor:latest",
+      "--conf", "spark.kubernetes.driver.docker.image=spark-driver:latest",
+      "--conf", "spark.ssl.kubernetes.driversubmitserver.enabled=true",
+      "--conf", "spark.ssl.kubernetes.driversubmitserver.keyStore=" +
+        s"file://${keyStore.getAbsolutePath}",
+      "--conf", "spark.ssl.kubernetes.driversubmitserver.keyStorePassword=changeit",
+      "--conf", "spark.ssl.kubernetes.driversubmitserver.keyPassword=changeit",
+      "--conf", "spark.ssl.kubernetes.driversubmitserver.trustStore=" +
+        s"file://${trustStore.getAbsolutePath}",
+      "--conf", s"spark.ssl.kubernetes.driversubmitserver.trustStorePassword=changeit",
+      "--conf", "spark.kubernetes.submit.waitAppCompletion=false",
+      EXAMPLES_JAR_FILE.getAbsolutePath)
+    SparkSubmit.main(args)
+  }
+
+  test("Enable SSL on the driver submit server using PEM files") {
```

Comment:
Do you think testing that use case directly is important enough?  It's definitely one I think people (and us) will be using